src/library_webgl.js

```diff
@@ -1831,7 +1831,7 @@ var LibraryGL = {
   glBufferSubData: function(target, offset, size, data) {
 #if MAX_WEBGL_VERSION >= 2
     if (GL.currentContext.version >= 2) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
-      GLctx.bufferSubData(target, offset, HEAPU8, data, size);
+      if (size) GLctx.bufferSubData(target, offset, HEAPU8, data, size);
```

Comment:
Also here can maybe understand this if check - however I wish browsers could handle this problem in the spec? Now we place a if check here because the browser behavior is not well defined?

Could this instead add a WebGL 2 conformance test to a defined behavior? (What I'd define the behavior to is to validate + then do a no-op if target and offset all both good)

Reply:
The behavior is well specified and tested in WebGL 2.0: https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.3 again. The issue is that the default size of "0" in WebGL is defined as uploading the entire passed ArrayBufferView, rather than a sub-region. C++ applications expect different behavior.